x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/middleware.ts

```diff
@@ -45,5 +45,23 @@ export const alertMiddlewareFactory: MiddlewareFactory<AlertListState> = (coreSt
       );
       api.dispatch({ type: 'serverReturnedAlertDetailsData', payload: response });
     }
+
+    if (action.type === 'userClosedAlert') {
+      const id = action.payload;
+      const body = JSON.stringify({
+        state: {
+          active: false,
+        },
+      });
+
+      try {
+        await coreStart.http.patch(`/api/endpoint/alerts/${id}`, {
```

Comment:
Are you going to be adding the server side implementation to this PR?
I'm concerned about using `patch` here. It might be nice to treat alerts as totally immutable and instead model the closing of an alert as a distinct entity. Let me know what you think about this.

Reply:
i believe the approach we're taking is to keep the alerts themselves immutable, the `state` obj is added onto that obj and is the only thing ever changed. It's how we're keeping track of things like open/closed state and the up to date host info for the alerts. @marshallmain might be able to expand on that more